shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+(function (global) {
+  'use strict';
+
+  var softKeyModule = (function () {
+
+    // Keys which are mapped to accesskey
+    const SK_KEYS_MAPPING = {
+      LSK: 'a',
+      CSK: 's',
+      RSK: 'd'
+    };
+
+    // Storing the reference to softkeys
+    var sk_keys = {
+      LSK: null,
+      CSK: null,
+      RSK: null
+    };
+
+    // Container created for SoftKey
+    var sk_container = null;
+
+    // Check if menu open
+    var sk_options_menu = {
+      open: false,
+      cancelKey: SK_KEYS_MAPPING.LSK,
+      cancelKeyText: 'Cancel',
+      cancelKeyl10nId: 'cancel'
+    };
+
+    // Observer for observing the DOM changes
+    var observer = null;
+
+    return {
+      init: init
+    };
+
+    /**
+     * Used to initialize the library
+     * @example
+     * softKeyModule.init();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function init() {
+
+      // Calling destroy to destroy previously initialized variables
+      destroy();
```

Comment:
I hope that this destroy() method is never actually needed. I'd hope that it would be initialized once, and then modified as needed after that.  

It can be useful to have init and destroy methods for unit tests, though..